src/world.cpp

```diff
@@ -236,59 +232,80 @@ bool World::is_over()const
-	{
-		m_traffic_cop.set_rotation(lanes_rot[1]);
-		m_lane_manager.turn_car(direction::WEST);
-	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_RIGHT) {
-		m_traffic_cop.set_rotation(lanes_rot[3]);
-		m_lane_manager.turn_car(direction::EAST);
-	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_W) {
-		m_lane_manager.input_create_cars(direction::NORTH);
+	if (action == GLFW_PRESS && key == GLFW_KEY_G && show_start_splash) { //start game with G
+		show_start_splash = !show_start_splash;
+		Mix_PlayMusic(m_game_music, -1);
 	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_A) {
-		m_lane_manager.input_create_cars(direction::WEST);
-	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_S) {
-		m_lane_manager.input_create_cars(direction::SOUTH);
-	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_D) {
-		m_lane_manager.input_create_cars(direction::EAST);
-	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_P) {
+	if (action == GLFW_PRESS && key == GLFW_KEY_P && !show_start_splash) { //pause anytime with P
 		is_game_paused = !is_game_paused;
+		is_game_paused ? Mix_PauseMusic() : Mix_ResumeMusic();
 	}
-	if (action == GLFW_PRESS && key == GLFW_KEY_SPACE) {
-		if (m_remove_intersection.show) {
-			m_remove_intersection.increment();
+	if (!is_game_paused && !show_start_splash) {
+		if (action == GLFW_PRESS && (key == GLFW_KEY_UP || key == GLFW_KEY_DOWN || key == GLFW_KEY_LEFT || key == GLFW_KEY_RIGHT)) {
+			Mix_PlayChannel(-1, m_roy_whistle, 0);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_UP) {
+			m_traffic_cop.set_rotation(lanes_rot[0]);
+			m_lane_manager.turn_car(direction::NORTH);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_DOWN) {
+			m_traffic_cop.set_rotation(lanes_rot[2]);
+			m_lane_manager.turn_car(direction::SOUTH);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_LEFT)
+		{
+			m_traffic_cop.set_rotation(lanes_rot[1]);
+			m_lane_manager.turn_car(direction::WEST);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_RIGHT) {
+			m_traffic_cop.set_rotation(lanes_rot[3]);
+			m_lane_manager.turn_car(direction::EAST);
 		}
-		if (m_remove_intersection.m_press == 10) {
-			clear_intersection();
+		if (action == GLFW_PRESS && key == GLFW_KEY_W) {
+			m_lane_manager.input_create_cars(direction::NORTH);
 		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_A) {
+			m_lane_manager.input_create_cars(direction::WEST);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_S) {
+			m_lane_manager.input_create_cars(direction::SOUTH);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_D) {
+			m_lane_manager.input_create_cars(direction::EAST);
+		}
+		if (action == GLFW_PRESS && key == GLFW_KEY_SPACE) {
+			if (m_remove_intersection.show) {
+				m_remove_intersection.increment();
+			}
+			if (m_remove_intersection.m_press == 10) {
+				clear_intersection();
+			}
+		}
+	}
+	if (action == GLFW_RELEASE && key == GLFW_KEY_R)
+	{
+		int w, h;
+		glfwGetWindowSize(m_window, &w, &h);
+
+		//TODO: All the destroys to free up memory

```

Comment:
Were you thinking of just calling all the destroys right before re-calling all the inits? or would that fuck up all the internal variables in the world class?

Reply:
Yeah calling all the destroys before calling the inits. That's how the salmon template did it.